proposals/0216-widget-support.md

```diff
@@ -246,73 +263,268 @@ The RPC `Show` and `SetDisplayLayout` are identified as the only screen related
+    <description>this RPC is deprecated in favor of SetScreenLayout</description>
+</function>
+
+<function name="SetScreenLayout" functionID="SetScreenLayoutID" messagetype="request" since="5.x">
+  <param name="screenLayout" type="String" maxlength="500" mandatory="true">
+    <description>
+        Predefined or dynamically created screen layout.
+        Currently only predefined screen layouts are defined.
+    </description>
+  </param>
+
+  <param name="dayColorScheme" type="TemplateColorScheme" mandatory="false" />
+  <param name="nightColorScheme" type="TemplateColorScheme" mandatory="false" />
+</function>
+
+<function name="SetScreenLayout" functionID="SetScreenLayoutID" messagetype="response" since="5.x">
+  <param name="info" type="String" maxlength="1000" mandatory="false" platform="documentation">
+      <description>Provides additional human readable info regarding the result.</description>
+  </param>
+  <param name="success" type="Boolean" platform="documentation" mandatory="true">
+    <description> true, if successful; false, if failed </description>
+  </param>
+  <param name="resultCode" type="Result" platform="documentation" mandatory="true">
+    <description>See Result</description>
+    <element name="SUCCESS"/>
+    <element name="INVALID_DATA"/>
+    <element name="OUT_OF_MEMORY"/>
+    <element name="TOO_MANY_PENDING_REQUESTS"/>
+    <element name="APPLICATION_NOT_REGISTERED"/>
+    <element name="GENERIC_ERROR"/>
+    <element name="REJECTED"/>
+    <element name="UNSUPPORTED_REQUEST"/>
+  </param>
+</function>
+```
+
+The information contained in the deprecated parameters will be made available with the newly proposed screen capability struct. In the next major release these parameters can be marked as removed.
+
+> Note: The alternative solution describe how to keep `SetDisplayLayout` and deprecate just the parameters within the response RPC. Above solution to create a new `SetScreenLayout` RPC is more clean to the naming and design, which is why it's the primary proposed solution. Either way works and can be chosen by the steering committee.
+
+#### Automatic subscription to display and screen capabilities
+
+As accepted in the app services proposal, the application can subscribe to system capabilities. With this proposal it would inclde display capabilities. In order to provide display capabilities as soon as possible after the app registered, the application should be automatically subscribed to display capabilities. With this rule, Core should send a system capability notification with display capabilities right after sending the response of the app registration. This approach results in a better performance compared to the need of the app to get/subscribe to display capabilities. Without this approach applications would perform slower as with the todays solution having capabilities being returned in `RegisterAppInterfaceResponse`.
```

Comment:
Update to:

As accepted in the app services proposal, the application can subscribe to system capabilities. With this proposal it would inclde display capabilities. In order to provide display capabilities as soon as possible after the app registered, the application should be automatically subscribed to display capabilities. With this rule, Core should send a system capability notification with display capabilities right after sending the response of the app registration. This approach **will** results in a better performance compared to the need of the app to get/subscribe to display capabilities. Without this approach, applications **will** perform slower than the **solution used today, which is having capabilities being returned in** `RegisterAppInterfaceResponse`.